supertokens_python/recipe/openid/utils.py

```diff
@@ -85,4 +97,28 @@ def validate_and_normalise_user_input(
         OverrideConfig(functions=override.functions, apis=override.apis),
         issuer_domain,
         issuer_path,
+        is_issuer_domain_given,
+    )
+
+
+async def get_issuer_domain_or_throw_error(
+    issuer_domain: Optional[str],
+    config: OpenIdConfig,
+    app_info: AppInfo,
+    user_context: Dict[str, Any],
+) -> NormalisedURLDomain:
+    if issuer_domain is not None:
+        return NormalisedURLDomain(issuer_domain)
+    if config.is_issuer_domain_given:
+        issuer_domain_resp = await config.issuer_domain(None, user_context)
+        return issuer_domain_resp
+    req = get_request_from_user_context(user_context)
+    if req is not None:
+        issuer_domain_resp = await config.issuer_domain(req, user_context)
+        return issuer_domain_resp
+    if app_info.initial_api_domain_type == "string":
+        issuer_domain_resp = await config.issuer_domain(None, user_context)
+        return issuer_domain_resp
+    raise Exception(
+        "Please pass issuer_domain as a string to the function or initiate issuer_domain with openid recipe or pass api_domain as string in supertokens.init"
     )
```

Comment:
this whole thing can be moved into the normalised version of issuer_domain, and then you also don't need is_issuer_domain_given and initial_api_domain_type